ir/pattern/constraint.rs

```diff
@@ -653,7 +661,9 @@ impl<ID: IrID> ExpressionBinding<ID> {
         F: FnMut(ID, ConstraintIDSide),
     {
         self.ids_assigned().for_each(|id| function(id, ConstraintIDSide::Left));
-        self.expression().ids().for_each(|id| function(id, ConstraintIDSide::Right));
+        // TODO
+        // todo!("Do we really need positions here?")
+        // self.expression().ids().for_each(|id| function(id, ConstraintIDSide::Right));
```

Comment:
I still don't see the need for anything in traversal to leak into the ir I like this.

Reply:
I'm going to try to get rid of it